src/libsyntax/source_map.rs

```diff
@@ -118,6 +118,18 @@ impl StableFilemapId {
 
         StableFilemapId(hasher.finish())
     }
+
+    pub fn new_from_pieces(name: &FileName,
+                           name_was_remapped: bool,
+                           unmapped_path: Option<FileName>) -> StableFilemapId {
```

Comment:
`unmapped_path` should be passed as a reference (either `&Option<FileName>` or `Option<&FileName>`). Then you don't have to clone it. You can use `Option::as_ref()` to convert from `Option<T>` to `Option<&T>`.